sklearn/pipeline.py

```diff
@@ -187,14 +188,10 @@ def _final_estimator(self):
     def _fit(self, X, y=None, **fit_params):
         self._validate_steps()
         # Setup the memory
-        memory = self.memory
-        if memory is None:
-            memory = Memory(cachedir=None, verbose=0)
-        elif isinstance(memory, six.string_types):
-            memory = Memory(cachedir=memory, verbose=0)
-        elif not isinstance(memory, Memory):
+        memory = check_memory(self.memory)
+        if not hasattr(memory, 'cache'):
```

Comment:
This raise statement will be duplicate. You need to include it in `check_memory`.